Makefile

```
@@ -69,7 +69,7 @@ PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
 CONTROLLER_GEN = go run ${PROJECT_DIR}/vendor/sigs.k8s.io/controller-tools/cmd/controller-gen
 ENVTEST = go run ${PROJECT_DIR}/vendor/sigs.k8s.io/controller-runtime/tools/setup-envtest
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
-ENVTEST_K8S_VERSION = 1.30.1
+ENVTEST_K8S_VERSION = 1.31.1
```

```suggestion
ENVTEST_K8S_VERSION = 1.31.0
```
This is the latest available in the upstream bucket.

Also at line 111 we need to get rid of the deprecated GCS flag and the openshift fork bucket. See https://github.com/openshift/cluster-control-plane-machine-set-operator/pull/328#discussion_r1773196811 for reference.